src/sound/SoundMusic.cpp

```diff
@@ -109,18 +89,7 @@ namespace Sound {
 
 	const std::vector<std::string> MusicPlayer::GetSongList() const
 	{
-		using std::pair;
-		using std::string;
-		const std::map<string, Sample> samples = Sound::GetSamples();
-		std::vector<std::string> songs;
-		songs.reserve(samples.size());
-		for (std::map<string, Sample>::const_iterator it = samples.begin();
-			 it != samples.end(); ++it) {
-			if (it->second.isMusic)
-				songs.emplace_back(it->first.c_str());
-		}
-
-		return songs;
+		return GetMusicFiles();
```

Comment:
This is conceptually messy, but in the interest of not increasing the scope of this PR I'll consider this "not worse than before"; the compiler is very likely to perform RVO here, so it's not a significant performance issue.